src/MealManager.js

```diff
@@ -0,0 +1,144 @@
+import _ from 'lodash';
+import Like from './Like.js';
+import Comment from './Comment.js';
+import { doGet, doPost } from './webRequest.js';
+
+// base APIs urls
+// Meal Database
+const mealAPI = 'https://www.themealdb.com/api/json/v1/1';
+const mealById = '/lookup.php?i=';
+// Involvement data storage
+const dataAPI = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/jv9JX1Y91XDdVtvfca13';
+const dataLikes = '/likes/';
+const dataComments = '/comments';
+const itemIdPostfix = '?item_id=';
+
+// Preserved Items
+const appData = [52772, 52773, 52774, 52775, 52776, 52777];
+
+export default class Meal {
+  static meals = [];
+
+  constructor(idMeal, strArea, strCategory, strInstructions,
+    strMeal, strMealThumb, strSource, strYoutube) {
+    this.idMeal = idMeal;
+    this.strArea = strArea;
+    this.strCategory = strCategory;
+    this.strInstructions = strInstructions;
+    this.strMeal = strMeal;
+    this.strMealThumb = strMealThumb;
+    this.strSource = strSource;
+    this.strYoutube = strYoutube;
+    this.comments = [];
+  }
+
+  static async GetMealRandom() {
+    const mealPromises = [];
+    appData.forEach((id) => {
+      const url = _.join([mealAPI, mealById, id], '');
+      mealPromises.push(
+        doGet(url).then((ans) => {
+          const tmp = ans.meals[0];
+          return new Meal(
+            tmp.idMeal,
+            tmp.strArea,
+            tmp.strCategory,
+            tmp.strInstructions,
+            tmp.strMeal,
+            tmp.strMealThumb,
+            tmp.strSource,
+            tmp.strYoutube,
+          );
+        }),
+      );
+    });
+    const mealsRandArray = await Promise.all(mealPromises);
+    this.meals = mealsRandArray;
+  }
+
+  static async GetMealId(idMeal) {
+    const url = _.join([mealAPI, mealById, idMeal], '');
+    const myMeal = await doGet(url).then((ans) => {
+      const tmp = ans.meals[0];
+      return new Meal(tmp.idMeal, tmp.strArea, tmp.strCategory, tmp.strInstructions, tmp.strMeal,
+        tmp.strMealThumb, tmp.strSource, tmp.strYoutube);
+    });
+    return myMeal;
+  }
+
+  static async GetLikes() {
+    const url = _.join([dataAPI, dataLikes], '');
+    const tmpAns = await doGet(url);
+    try {
+      tmpAns.forEach((lk) => {
+        const tmpLike = new Like(lk.item_id, lk.likes);
+        Like.likes.push(tmpLike);
+      });
+    } catch (e) {
+      console.log(`GetLikes error: ${e}`);
```

Comment:
- [ ] Please remove the `console` code as it is affecting performance and it's not code-related but debugging only 👍 see the hint [here](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md#:~:text=Do%20not%20commit%20console.log)